### DISCUSSION TITLE
LIS run error

### DISCUSSION THREAD

**Discussion by Giamik62** [NASA-LIS/LISF#975] [Q&A] [answered]
Hi, lisf community,
I have intel parallel studio v2019u5 (v2019.5.281) on my cluster.  I successfully compiled the latest version of LISF (LIS) with this compiler.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
export LIS_ARCH=linux_ifc
export LIS_FC=mpiifort
export LIS_CC=mpiicc
export LIS_MODESMF=/scratch/LandSurface_Models/esmf/mod/modg/Linux.intel.64.mpiuni.default
export LIS_LIBESMF=/scratch/LandSurface_Models/esmf/lib/libg/Linux.intel.64.mpiuni.default
export LIS_OPENJPEG={v2.4.0} LIS_ECCODES={v2.12.5} LIS_NETCDF={c - v4.7.4, fortran - v4.5.3, cxx - v4.4.3.1} LIS_HDF4={v2.15} LIS_HDF5={v1.12.0} LIS_HDFEOS={v2.20}
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
The ESMF version is 8.2.0. All these libraries are also compiled with the same intel compilers.
When I run LIS program
mpirun  -n 64 ./LIS
it crashes with error: Attempting to use an MPI routine before initializing MPICH
No matter runs LIS in parallel or sequential mode!
Can anyone help me?

**Comment by bmcandr** [comment, marked as answer]
Hi @Giamik62,

I notice that the paths in your `LIS_MODESMF` and `LIS_LIBESMF` variables includes "mpiuni". This indicates that ESMF was built with the MPI bypass library option:

>Alternatively, ESMF comes with a single-processor MPI-bypass library which is the default for Linux and Darwin systems. To force the use of this bypass library set ESMF_COMM equal to "mpiuni".
* [See `ESMF_COMM` in the ESMF Environment Variables section](https://earthsystemmodeling.org/docs/release/latest/ESMF_usrdoc/node10.html#EnvironmentVariables)

You will need to link ESMF to your system's MPI installation during the build process per the instructions at the above link and update the paths in `LIS_MODESMF` and `LIS_LIBESMF` accordingly.

----

I reproduced the error message you reported on our system by pointing `LIS_MODESMF` and `LIS_LIBESMF` to our "mpiuni" build of ESMF so I'm fairly confident this is the issue.

**Comment by Giamik62** [comment]
Many thanks, bmcandr,
I have changed the value of the parameter of ESMF_COMM on 'intelmpi' and the problem was disappeared!

Best regards
George

**Reply by bmcandr** [reply]
Great!
